pynest/examples/Potjans_2014/network.py

```diff
@@ -0,0 +1,475 @@
+                self.poisson[i], target_pop,
+                conn_spec=conn_dict_poisson,
+                syn_spec=syn_dict_poisson
+                )
+
+    def connect_thalamus(self):
+        """ Connects the thalamic population to the microcircuit."""
+        print('Thalamus connection established')
+        for i, target_pop in enumerate(self.pops):
+            conn_dict_th = {
+                'rule': 'fixed_total_number',
+                'N': int(self.nr_synapses_th[i])
+                }
+            syn_dict_th = {
+                'weight': {
+                    'distribution': 'normal_clipped',
+                    'mu': self.thalamic_weight,
+                    'sigma': (
+                        self.thalamic_weight * self.net_dict['PSP_sd']
+                        ),
+                    'low': 0.0
+                    },
+                'delay': {
+                    'distribution': 'normal_clipped',
+                    'mu': self.stim_dict['delay_th'][i],
+                    'sigma': self.stim_dict['delay_th_sd'][i],
+                    'low': self.sim_resolution
+                    }
+                }
+            nest.Connect(
+                self.thalamic_population, target_pop,
+                conn_spec=conn_dict_th, syn_spec=syn_dict_th
+                )
+
+    def connect_dc_generator(self):
+        """ This function connects the DC generator to the microcircuit."""
+        print('DC Generator connection established')
+        for i, target_pop in enumerate(self.pops):
+            if self.stim_dict['dc_input']:
+                nest.Connect(self.dc[i], target_pop)
+
+    def connect_devices(self):
+        """ Connects the recording devices to the microcircuit."""
+        print('%s of 2 Devices connected' % (len(self.net_dict['rec_dev'])))
+        for i, target_pop in enumerate(self.pops):
+            if 'voltmeter' in self.net_dict['rec_dev']:
+                nest.Connect(self.voltmeter[i], target_pop)
+            if 'spike_detector' in self.net_dict['rec_dev']:
+                nest.Connect(target_pop, self.spike_detector[i])
+
+    def setup(self):
+        """ Execute subfunctions of the network.
+
+        This function executes several subfunctions to create neuronal
+        populations, devices and inputs, connects the populations with
+        each other and with devices and input nodes.
+
+        """
+        self.setup_nest()
+        self.create_populations()
+        self.create_devices()
+        self.create_thalamic_input()
+        self.create_poisson()
+        self.create_dc_generator()
+        self.create_connections()
+        if self.net_dict['poisson_input']:
+            self.connect_poisson()
+        if self.stim_dict['thalamic_input']:
+            self.connect_thalamus()
+        if self.stim_dict['dc_input']:
+            self.connect_dc_generator()
+        self.connect_devices()
+
+    def simulate(self):
+        """ This function starts the simulation."""
```

Comment:
It does not only start the simulation, thus it should read e.g.
""" This function simulates the network."""